packages/debug/src/browser/debug-session.tsx

```diff
@@ -302,22 +311,27 @@ export class DebugSession implements CompositeTreeElement {
             await this.disconnect(restart);
         }
     }
+
     protected async disconnect(restart?: boolean): Promise<void> {
-        try {
-            await this.sendRequest('disconnect', { restart });
-        } catch (reason) {
-            this.fireExited(reason);
-            return;
-        }
-        const timeout = 500;
-        if (!await this.exited(timeout)) {
-            this.fireExited(new Error(`timeout after ${timeout} ms`));
-        }
+        const TIMEOUT_MS = 1000;
+        Promise.race([
+            this.sendRequest('disconnect', { restart }),
+            new Promise(reject => setTimeout(reject, TIMEOUT_MS, new Error('TIMEOUT_ERR')))
+        ]).then(res => {
```

Comment:
you can use `await` and `catch` instead of `.then().catch()`

Reply:
I tried it and other options but found that setTimout throwing exception in a promise is not triggering catch but returning result - which is an error.
Please see https://javascript.info/task/error-async